plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/CubridStructureAssistant.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.cubrid.model;
+
+import java.sql.SQLException;
+import java.util.List;
+
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.ext.cubrid.CubridConstants;
+import org.jkiss.dbeaver.ext.generic.model.GenericSchema;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCPreparedStatement;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCResultSet;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCSession;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCExecutionContext;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCStructureAssistant;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCUtils;
+import org.jkiss.dbeaver.model.impl.struct.AbstractObjectReference;
+import org.jkiss.dbeaver.model.impl.struct.RelationalObjectType;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.model.struct.DBSObject;
+import org.jkiss.dbeaver.model.struct.DBSObjectReference;
+import org.jkiss.dbeaver.model.struct.DBSObjectType;
+
+public class CubridStructureAssistant extends JDBCStructureAssistant<JDBCExecutionContext> {
+
+    private final CubridDataSource dataSource;
+
+    public CubridStructureAssistant(CubridDataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    @Override
+    public JDBCDataSource getDataSource() {
+        return dataSource;
+    }
+
+    @Override
+    protected void findObjectsByMask(
+            JDBCExecutionContext executionContext,
+            JDBCSession session,
+            DBSObjectType objectType,
+            ObjectsSearchParams params,
+            List<DBSObjectReference> references)
+            throws DBException, SQLException {

```

Comment:
Please pay attention to the code style guidelines, especially to the formatting section https://github.com/dbeaver/dbeaver/wiki/Contribute-your-code#formatting . We use 4-spaces formatting everywhere.
```suggestion
    protected void findObjectsByMask(
        JDBCExecutionContext executionContext,
        JDBCSession session,
        DBSObjectType objectType,
        ObjectsSearchParams params,
        List<DBSObjectReference> references
    ) throws DBException, SQLException {

```